### ISSUE TITLE
module map file '/Users/***/ios/Pods/Headers/Public/flutter_plugin_android_lifecycle/flutter_plugin_android_lifecycle.modulemap' not found

### ISSUE THREAD

**Issue by jianhuaz** [flutter/flutter#76593]
flutter run --release -d ***
fatal error: module map file '/Users/***/ios/Pods/Headers/Public/flutter_plugin_android_lifecycle/flutter_plugin_android_lifecycle.modulemap' not found
1 error generated.

pubspec.yaml
image_picker: 0.6.3+1
flutter_plugin_android_lifecycle: 1.0.5

flutter doctor
[✓] Flutter (Channel unknown, 1.20.4, on macOS 11.1 20C69, locale en-CN)

[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
[✓] Xcode - develop for iOS and macOS (Xcode 12.4)
[✓] Android Studio (version 4.1)
[✓] Connected device (1 available)
• No issues found!

**Comment by darshankawar** [comment]
@jianhuaz 
You seem to be on older version of Stable. Current stable is 1.22.6. I recommend you to upgrade to latest stable and try again to see if it works.
If issue still persists, provide updated `flutter doctor -v`, `pubspec.yaml` and a [complete minimal reproducible code sample](https://stackoverflow.com/help/minimal-reproducible-example) that shows the issue.

**Comment by jianhuaz** [comment]
@darshankawar

$ flutter run
...
 MTBBarcodeScanner-adadwpfuphqtfbaicqchvflqdulg
    9 warnings generated.
    fatal error: module map file '/Users/***/ios/Pods/Headers/Public/flutter_plugin_android_lifecycle/flutter_plugin_android_lifecycle.modulemap' not found
    1 error generated.
    Pods-Runner-adadwpfuphqtfbaicqchvflqdulg
    note: Using new build system
    note: Building targets in parallel
    note: Planning build
    note: Constructing build description
...

$ flutter doctor -v
[✓] Flutter (Channel unknown, 1.22.6, on macOS 11.1 20C69 darwin-x64, locale en-CN)
    • Flutter version 1.22.6 at /Users/jhzhou/Tools/flutter
    • Framework revision 9b2d32b605 (5 weeks ago), 2021-01-22 14:36:39 -0800
    • Engine revision 2f0af37152
    • Dart version 2.10.5
    • Pub download mirror https://pub.flutter-io.cn
    • Flutter download mirror https://storage.flutter-io.cn

 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at /Users/jhzhou/Library/Android/sdk
    • Platform android-29, build-tools 29.0.2
    • ANDROID_SDK_ROOT = /Users/jhzhou/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 12.4)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.4, Build version 12D4e
    • CocoaPods version 1.10.1

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin installed
    • Dart plugin version 201.9317
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)

[✓] Connected device (1 available)
    • jianhua的 iPhone (mobile) • 9497aef6cc02d6e945b9b620d15e93f4b9d611a0 • ios • iOS 12.4.9
    ! Error: jianhua的 iPhone is busy: Copying cache files from device. Xcode will continue when jianhua的 iPhone is finished. (code -10)
    ! Error: Jianhua iPad is not connected. Xcode will continue when Jianhua iPad is connected. (code -13)

• No issues found!


#pubspec.yaml

dependencies:
  flutter:
    sdk: flutter
  flutter_localizations:
    sdk: flutter

  date_format: ^1.0.8
  qr_flutter: ^3.2.0

  jpush_flutter: ^0.5.5
  path_provider: ^1.6.11
  share_extend: ^1.0.7

#  flutter_tts:
  zefyr: ^0.10.0
  orientation: ^1.2.0
  amap_location:
  flutter_echarts:
  lpinyin:
  package_info: ^0.4.0
  #  async_loader:
  json_serializable: ^3.2.2

  oktoast: ^2.3.1
  dio: ^3.0.9
  rxdart: ^0.23.1
  common_utils: ^1.1.3
  flustars: 0.2.6+1
  flutter_swiper: ^1.1.6
  url_launcher: 5.4.1
  image_picker: 0.6.7+22
  flutter_plugin_android_lifecycle: 1.0.11
  flutter_slidable: ^0.5.4
  sticky_headers: ^0.1.8+1
  azlistview: ^0.1.2
  simple_gesture_detector: ^0.1.4
  fluro: ^1.5.2
  cached_network_image: ^2.0.0
  sprintf: ^4.0.2
  keyboard_actions: 3.2.0
  provider: ^4.0.4
  barcode_scan:
    git:
      url: https://github.com/simplezhli/flutter_barcode_reader.git

**Comment by jianhuaz** [comment]
@darshankawar 

flutter version 1.17.5

pubspec.yaml
image_picker: 0.6.3+1
flutter_plugin_android_lifecycle: 1.0.5

is OK

**Comment by darshankawar** [comment]
@jianhuaz 
Is your `podfile` in sync with all plugin versions ?
In my local app, I don't have any folders inside `Headers` under ios/podfile/headers`, so wondering how you have got `public` folder there.

Can you try `flutter pub cache repair` and then `flutter run` to see if it helps ?

Assuming you are in china `Pub download mirror https://pub.flutter-io.cn
   • Flutter download mirror https://storage.flutter-io.cn`

Can you confirm if this is not an issue with `https://pub.flutter-io.cn` or you need to update the flutter mirror china ?

**Comment by jianhuaz** [comment]
@darshankawar 
Thank you. At present, our task is very tight. We will return to futter 1.17.5 to release the app, wait for the upgrade when we have time.

**Comment by darshankawar** [comment]
@jianhuaz 
Feel free to close this issue and reopen if required after you upgrade and still see the issue.

**Comment by pedromassangocode** [comment]
Without additional information, we are unfortunately not sure how to resolve this issue. 
We are therefore reluctantly going to close this bug for now. 
Please don't hesitate to comment on the bug if you have any more information for us; we will reopen it right away!  
Thanks for your contribution.

Could everyone who still has this problem please file a new issue with the exact description of what happens, logs, and the output of `flutter doctor -v`.
All system setups can be slightly different, so it's always better to open new issues and reference related issues.

**Comment by pedromassangocode** [comment]
Without additional information, we are unfortunately not sure how to resolve this issue. 
We are therefore reluctantly going to close this bug for now. 
Please don't hesitate to comment on the bug if you have any more information for us; we will reopen it right away!  
Thanks for your contribution.

Could everyone who still has this problem please file a new issue with the exact description of what happens, logs, and the output of `flutter doctor -v`.
All system setups can be slightly different, so it's always better to open new issues and reference related issues.

**Comment by jianhuaz** [comment]
flutter doctor
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel unknown, 2.0.1, on macOS 11.3.1 20E241 darwin-x64, locale en-CN)
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
[✓] Xcode - develop for iOS and macOS
[✓] Chrome - develop for the web
[✓] Android Studio (version 4.1)
[✓] Connected device (2 available)
    ! Error: jianhua的 iPhone is not connected. Xcode will continue when jianhua的 iPhone is connected. (code -13)
    ! Error: Jianhua iPad is not connected. Xcode will continue when Jianhua iPad is connected. (code -13)

• No issues found!


pubspec.yaml
...
image_picker: ^0.8.0+3
...

image_picker pubspec.yaml
...
flutter_plugin_android_lifecycle: ^2.0.1
...

flutter_plugin_android_lifecycle  pubspec.yaml
...
flutter:
  plugin:
    platforms:
      android:
        package: io.flutter.plugins.flutter_plugin_android_lifecycle
        pluginClass: FlutterAndroidLifecyclePlugin
...

flutter run (iOS)
fatal error: module map file '/Users/***/ios/Pods/Headers/Public/flutter_plugin_android_lifecycle/flutter_plugin_android_lifecycle.modulemap' not found

Because flutter_plugin_android_lifecycle plugin has no IOS platform, it will report this error. How do I need to use image_picker?
